modules/react/loading-sparkles/lib/LoadingSparkles.tsx

```diff
@@ -0,0 +1,168 @@
+ */
+const createAnimation = (stages: CSSObject[]) =>
+  keyframes({
+    '0%': stages[2],
+    '20%': stages[2],
+    '47%': stages[0],
+    '73%': stages[1],
+    '100%': stages[2],
+  });
+
+/**
+ * Creates a keyframe animation that cycles through colors for each stage.
+ */
+const COLOR_ANIMATION = createAnimation([
+  {fill: AI_COLORS.dragonFruit400},
+  {fill: AI_COLORS.dragonFruit300},
+  {fill: AI_COLORS.dragonFruit200},
+]);
+
+/**
+ * Creates a keyframe animation that scales the sparkle to the given stage.
+ */
+const createScaleAnimation = (stageIndex: number, stageScales: number[]) =>
+  createAnimation(
+    stageScales.map((scale, index) => ({
+      opacity: index === stageIndex ? 1 : 0,
+      transform: `scale(${scale})`,
+    }))
+  );
+
+type SparkleAnimationIconProps = {
+  /**
+   * The animation delay of the sparkle (in ms).
+   */
+  animationDelay: number;
+
+  /**
+   * At which stage the icon is visible.
+   */
+  stageIndex: number;
+
+  /**
+   * The scales of the sparkle at each stage.
+   */
+  stageScales: number[];
+} & React.ComponentProps<typeof SystemIcon>;
+
+/**
+ * The animated sparkle icon.
+ */
+const SparkleAnimationIcon = styled(SystemIcon)<SparkleAnimationIconProps>(
+  {
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    animationDuration: ANIMATION_DURATION_MS + 'ms',
+    animationFillMode: 'both',
+    animationIterationCount: 'infinite',
+    '.wd-sparkle-fill': {
+      animationDuration: ANIMATION_DURATION_MS + 'ms',
+      animationFillMode: 'both',
+      animationIterationCount: 'infinite',
+      animationName: COLOR_ANIMATION,
+    },
+  },
+  ({animationDelay, stageIndex, stageScales}) => ({
+    animationDelay: animationDelay + 'ms',
+    animationName: createScaleAnimation(stageIndex, stageScales),
+    '.wd-sparkle-fill': {
+      animationDelay: animationDelay + 'ms',
+    },
+  })
+);
+
+/**
+ * A container for the loading sparkle stages.
+ */
+const SparkleContainer = styled('div')({
+  position: 'relative',
+  width: canvas.space.xs,
+  height: canvas.space.xs,
+});
+
+type SparkleProps = {
+  /**
+   * The animation delay of the sparkle (in ms).
+   */
+  animationDelay?: number;
+};
+
+/**
+ * A loading sparkle, which consists of three SVGs laid on top of each other.
+ */
+const Sparkle = ({animationDelay = 0}: SparkleProps) => (
+  <SparkleContainer>
+    <SparkleAnimationIcon
+      icon={sparkleStage0Icon}
+      stageIndex={0}
+      stageScales={[1, 0.7, 0.55]}
+      animationDelay={animationDelay}
+      size={canvas.space.xs}
+    />
+    <SparkleAnimationIcon
+      icon={sparkleStage1Icon}
+      stageIndex={1}
+      stageScales={[1.4, 1, 0.8]}
+      animationDelay={animationDelay}
+      size={canvas.space.xs}
+    />
+    <SparkleAnimationIcon
+      icon={sparkleStage2Icon}
+      stageIndex={2}
+      stageScales={[1.8, 1.25, 1]}
+      animationDelay={animationDelay}
+      size={canvas.space.xs}
+    />
```

Comment:
Why are there 3 stacked icons? From what I can tell from the Figma animation, there's a single sparkle that grows and shrinks and changes color. Similar to the loading dots animation.

If the animation can be reduced to a single icon, this file can be greatly simplified. As it is right now, there's a lot of complicated abstractions that will be extremely difficult to refactor to static styling.

Reply:
There are slight differences between the three stages (the smaller stages are less "pointy" than the largest) – that being said, given the scale of the animation (12px high) your suggestion would simplify this greatly.

Reply:
I have an example sandbox. At 12px, it isn't possible to see the radius difference: https://codesandbox.io/p/sandbox/jolly-driscoll-84q9kf?file=%2Fsrc%2FLoadingSparkles.tsx%3A34%2C29